R/assertions.R

```diff
@@ -1527,13 +1562,14 @@ assert_list_element <- function(list,
       )
       message <- c(
         message_text,
-        i = paste(" But,", info_msg)
+        i = paste("But,", info_msg)
       )
     }
 
     cli_abort(
       message = message,
       class = c(class, "assert-admiraldev"),
+      .envir = env(...),
```

Comment:
Are we setting up `cli_abort` correctly?  Apologies I am terrible at this stuff but feels a little funky 
![image](https://github.com/user-attachments/assets/eb79372f-1377-4922-99ec-2c4fede6d38f)

Reply:
When we started, it was suggested not to include an `envir` argument for simplicity. In the generic checking functions I maintain, I prefer to expose the environment control just in case, for example:

```r
check_class <- function(x,
                        cls,
                        allow_empty = FALSE,
                        message,
                        arg_name = rlang::caller_arg(x),
                        class = "check_class",
                        call = get_cli_abort_call(),
                        envir = rlang::current_env()) {  # ENVIR ARGUMENT IS HERE AND IS PASSED TO cli_abort()
  # if empty, skip test
  if (isTRUE(allow_empty) && rlang::is_empty(x)) {
    return(invisible(x))
  }

  if (!inherits(x, cls)) {
    cli::cli_abort(message, class = c(class, "standalone-checks"), call = call, .envir = envir)
  }
  invisible(x)
}
```

We could do the same here

Reply:
I think this would work for `assert_list_element()`. However, if we add the `envir` argument here, we should add it for all assertions. This would be a lot of work. We would not only need to add the argument but also revise and update the code of the assertions because in some assertions the message text uses objects which are created in the assertion function.
Thus I would not add the `envir` argument.

Reply:
okay sounds good to not add